db/R/dbfiles.R

```diff
@@ -65,12 +66,53 @@ dbfile.input.insert <- function(in.path, in.prefix, siteid, startdate, enddate,
     db.query(cmd, con)
     # return input id
     inputid <- db.query(paste0("SELECT id FROM inputs WHERE site_id=", siteid, " AND format_id=", formatid, " AND start_date='", startdate, "' AND end_date='", enddate, "'" , parent, ";"), con)[['id']]
+  } else {
+    if(nrow(existing.input) > 1) {
+      print(existing.input)
+      logger.warn("Multiple existing inputs found. Using last.")
+      existing.input <- existing.input[nrow(existing.input),]
+    }
+    
+    # Convert dates to Date objects and strip all time zones (DB values are timezone-free)
+    startdate <- force_tz(as_date(startdate), 'GMT')
+    enddate <- force_tz(as_date(enddate), 'GMT')
+    existing.input$start_date <- force_tz(as_date(existing.input$start_date), 'GMT')
+    existing.input$end_date <- force_tz(as_date(existing.input$end_date), 'GMT')
+    
+    if(existing.input$start_date != startdate || existing.input$end_date != enddate) {
+      print(existing.input, digits=10)
+      logger.error(paste0(
```

Comment:
Most of the changes in dbfiles.R, aside from cosmetics, added errors like this one. I know it might not be the intent to be so strict, in which case abandon these. But I would say that the whole inputs/dbfiles thing could be a lot less confusing if it were more rigidly defined.